File: src/Parrot.Renderers/SelfClosingRenderer.cs
namespace Parrot.Renderers
{
    using System;
    using System.Collections.Generic;
    using Parrot.Infrastructure;
    using Parrot.Nodes;
    using Parrot.Renderers.Infrastructure;

    public class SelfClosingRenderer : HtmlRenderer
    {
        public SelfClosingRenderer(IHost host) : base(host)
        {
        }

        public override IEnumerable<string> Elements
        {
            get { return new[] {"base", "basefont", "frame", "link", "meta", "area", "br", "col", "hr", "img", "param"}; }
        }

        public override void Render(IParrotWriter writer, IRendererFactory rendererFactory, Statement statement, IDictionary<string, object> documentHost, object model)
        {
            var localModel = GetLocalModel(documentHost, statement, model);
            CreateTag(writer, rendererFactory, documentHost, localModel, statement);
        }

        protected override void CreateTag(IParrotWriter writer, IRendererFactory rendererFactory, IDictionary<string, object> documentHost, object model, Statement statement)
        {
            string tagName = string.IsNullOrWhiteSpace(statement.Name) ? DefaultChildTag : statement.Name;

            TagBuilder builder = new TagBuilder(tagName);
            //add attributes
            RenderAttributes(rendererFactory, documentHost, model, statement, builder);

            writer.Write(builder.ToString(TagRenderMode.SelfClosing));
        }
    }
}

File: src/Parrot.Javascript/scripts/Renderers/selfClosingRenderer.ts
///<reference path="../Parser/parser.ts" />
///<reference path="./irenderer.ts" />
///<reference path="rendererProvider.ts" />
///<reference path="../Infrastructure/ObjectModelValueProvider.ts" />
///<reference path="../Infrastructure/ValueTypeProvider.ts" />
///<reference path="../exceptions.ts" />
///<reference path="./htmlRenderer.ts" />

class SelfClosingRenderer extends HtmlRenderer {
    defaultChildTag: string = "";
    rendererProvider: RendererProvider;
    elements: string[] = ["base", "basefont", "frame", "link", "meta", "area", "br", "col", "hr", "img", "param"];

    render(statement: Statement, host: any[], model: any, rendererProvider: RendererProvider): string {
        //get the local model
        var localModel = this.getLocalModelValue(host, statement, model);

        return this.createTag(statement, host, localModel);
    }

    createTag(statement: Statement, host: any[], model: any): string {
        var tagName: string = statement.name == null || statement.name.length == 0 ? this.defaultChildTag : statement.name;
        var builder = new TagBuilder(tagName);
        this.renderAttributes(host, model, statement, builder);

        return builder.toString(TagRenderMode.selfClosing);
    }
}